framework/behaviors/SluggableBehavior.php

```diff
@@ -171,6 +171,10 @@ protected function isNewSlugNeeded()
             return false;
         }
 
+        if ($this->attribute === null) {
```

Comment:
> Can you support this with a comment to explain why true is to be returned when attribute is null?
Yes I can. And I did it here: https://github.com/yiisoft/yii2/pull/13729/files#r106027562

> What would be the usecase for that?
String-based slugs.

> What is the meaning of it. Why is it added?
It seems to be obvious for me. It's not possible to tell if new slug is needed or not for callback-based slugs. That is why a new value is always needed for such slugs.